src/metakb/harvesters/civic.py

```diff
@@ -35,50 +35,24 @@ def __init__(
 
         civicpy.load_cache(local_cache_path=local_cache_path, on_stale="ignore")
 
-        self.genes = []
-        self.variants = []
-        self.molecular_profiles = []
-        self.evidence = []
-        self.assertions = []
-
-    def harvest(self, harvested_filepath: str | None = None) -> bool:
-        """Retrieve and store evidence, gene, variant, molecular profile, and assertion
-        records from CIViC in composite and individual JSON files.
-
-        :param harvested_filepath: Path to the JSON file where the harvested data will
-            be stored. If not provided, will use the default path of
-            ``<APP_ROOT>/data/civic/harvester/civic_harvester_YYYYMMDD.json``
-        :return: `True` if operation was successful, `False` otherwise.
-        :rtype: bool
+    def get_harvester_data(self) -> dict:
+        """Get CIViC evidence, gene, variant, molecular profile, and assertion data
+
+        :return: Dictionary containing CIViC evidence items, genes, variants, molecular
+            profiles, and assertions
         """
-        try:
-            self.evidence = self.harvest_evidence()
-            self.genes = self.harvest_genes()
-            self.variants = self.harvest_variants()
-            self.molecular_profiles = self.harvest_molecular_profiles()
-            self.assertions = self.harvest_assertions()
-
-            json_created = self.create_json(
-                {
-                    "evidence": self.evidence,
-                    "genes": self.genes,
-                    "variants": self.variants,
-                    "molecular_profiles": self.molecular_profiles,
-                    "assertions": self.assertions,
-                },
-                harvested_filepath,
-            )
-            if not json_created:
-                logger.error(
-                    "CIViC Harvester was not successful: JSON files not created."
-                )
-                return False
-        except Exception as e:
-            logger.error("CIViC Harvester was not successful: %s", e)
-            return False
-        else:
-            logger.info("CIViC Harvester was successful.")
-            return True
+        evidence = self.harvest_evidence()
+        genes = self.harvest_genes()
+        variants = self.harvest_variants()
+        molecular_profiles = self.harvest_molecular_profiles()
+        assertions = self.harvest_assertions()
+        return {
+            "evidence": evidence,
+            "genes": genes,
+            "variants": variants,
+            "molecular_profiles": molecular_profiles,
+            "assertions": assertions,
+        }
```

Comment:
out of scope for this PR IMO, but one thing that makes some of the higher metakb methods so messy is how much stuff is built into deeply nested dicts. I wonder if it'd be better to rely on namedtuples/pydantic classes to enforce structure on objects passed around internally, instead of just waiting until the web API layer to put them through the pydantic classes.

Reply:
Does it make sense for us to use Pydantic classes for harvested unstructured data?

Reply:
I don't think you can for the stuff you're ingested, since its structure could just be whatever... on the other hand, the harvester data object is technically a dict but it's a dict with specific field names (and we wouldn't expect a consumer to alter those field names) so you could just make it a namedtuple or pydantic class so that those field names are known in advance

Reply:
Okay, that makes more sense. I thought you were suggesting pydantic models for the unstructured data. That's an easy change, so I can just include it in this PR